common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -365,7 +365,7 @@ protected boolean runAllTasks(long timeoutNanos) {
             // Check timeout every 64 tasks because nanoTime() is relatively expensive.
             // XXX: Hard-coded value - will make it configurable if it is really a problem.
             if ((runTasks & 0x3F) == 0) {
-                lastExecutionTime = ScheduledFutureTask.nanoTime();
+                lastExecutionTime = System.nanoTime();
                 if (lastExecutionTime >= deadline) {
```

Comment:
good catch.

@normanmaurer - i guess we need some work for expiration checks (something with the properties of `abs(t1 - t2) >= expire` to use raw `nanoTime()`?